Adjust Version Table Migration to fill initial data

meksor opened the pull request:
Currently the migration added in #166 will only work correctly if used on a fresh database. 
This PR adjusts the migration to copy in initial data from the corresponding table. 

I am currently still figuring out how to best build a test-setup for the migration. Ideas and considerations encouraged.

glatterf42 reviewed (approved):
Generally looks good to me, just one question that should not stop this.

As for migration tests, I'm thinking it should be possible to write a test that adds a new DB using ixmp4's CLI, then runs `alembic upgrade head` to perform the migrations, and invokes `alembic history` somehow to make sure `head` is at the latest migration hash code. 
But this seems fairly straightforward, so I'm guessing you considered this already: what's the issue with this that I'm missing?

meksor commented:
Yeah, if we want to just make sure that the migrations emit valid SQL that's an option. I was thinking about tests that actually check if /data/ is migrated correctly, which would mean that the tests need to load some test data every time a new table is created and afterwards check if the right data ended up in the right places...

glatterf42 commented on pyproject.toml:
```diff
@@ -26,7 +26,7 @@ httpx = { extras = ["http2"], version = ">=0.25.0" }
 # remove legacy-handling in timeseries- and meta-repositories when dropping pandas < 2.2
 pandas = ">=2.1.0"
-pandera = ">=0.17.0"
+pandera = ">=0.17.0, <0.24.0"
```
Okay, but why are we insisting on pandas 2.1.0? Could we not bump this to 2.1.1 and avoid the pin? The [commit](https://github.com/iiasa/ixmp4/commit/769a1985caf335b91b03370beff17bb2fcf3db86) setting the test for 2.1.0 seems to just have bumped the version for some reason, but not set it for any particular reason, so I'd try to bump pandas instead of pinning pandera.

meksor replied:
I guess this was the original origin :)
https://github.com/iiasa/ixmp4/pull/80

glatterf42 replied:
Unless we know pandera plans to remove that version bump with the next patch release or so.

glatterf42 commented:
Hm, then I misunderstood migrations. I thought these were just concerned with the structure of the DB, not the actual data. Testing the flow of data does sound more involved, but maybe we could also start with a test that checks that _any_ data is present after the migration (trusting/saving a more involved test for later)?

meksor commented:
Ok that was a doozey... first I added a dependency to test all the migrations in different ways (consistency with current models, upward-downward consistency, etc.), which prompted a first wave of easy bug fixes. 
Then, I adjusted the setup to test both the postgres and the sqlite database which revealed some more problems with the current migration setup.
PostgreSQL has a maximum limit for the length of names of constraints while SQLite has no such limit. This means we've been generating constraint names that are way too long and won't work on the prod DB.
To fix this, I added the `max_identifier_length` Param to all `create_engine` calls and stepped through the migrations to find the new constraint names. Took a few hours but most problems with the migrations should be fixed now and the tests should prevent further things from happening...
